### PR TITLE
Add tutorial: Supervised Learning with Hyperdimensional Computing

### PR DESCRIPTION
**Pull request by cumbof**
This PR adds a new tutorial titled **Supervised Learning with Hyperdimensional Computing** under the **Statistics and machine learning** topic. In particular, it focuses on how to use the [chopin2](https://github.com/cumbof/chopin2) tool in Galaxy for building a supervised classification model based on the hyperdimensional computing paradigm.

### CODE REVIEW

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  - "A vector-symbolic architecture is composed of vectors and a limited set of arithmentic operators"
+  - "Classification models build according to the hyperdimensional computing paradigm can scale on datasets with massive amounts of features and data points"
+contributors:
+  - cumbof
+
+---
+
+# Introduction
+
+`chopin2` ({% cite Cumbo2020 %}) implements a domain-agnostic supervised classification method based on the hyperdimensional (HD) computing paradigm. It is ano open-source tool and its code is available on GitHub at [https://github.com/cumbof/chopin2](https://github.com/cumbof/chopin2).
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:29]: It is __an__ open source tool

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Finally, the permutation operation is: (i) invertible, (ii) it distributes over bundling and any elementwise operation, (iii) it rpeserves the distance, and (iv) the resulting vector is dissimilar to the input vectors.
+
+The set of vectors and arithmetic operators used for combining vectors is called vector-symbolic architecture.
+
+`chopin2` implements a supervised classification model that works by encoding every observation in a dataset into high-dimensional vectors by combining values under their features. The model is built by collapsing the vector representation of the observations in the training set into a vector for each class of observations (e.g., in the case of the datasets retrieved in the previous step, classes are `CRC` and `control`). The classification model os then tested by computing the inner product between the vector representations of the observations in the test set against the two classes vectors. Vectors are thus classified based on their closeness to the classes.
+
+> <hands-on-title>Build a classification model with `chopin2`</hands-on-title>
+> **chopin2** {% icon tool %}: Run **chopin2** with the following configuration:
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:96]: You can annotate tools with Galaxy tool IDs. This then will yield of all kind of cool integrations.

Something like `{% tool [MultiQC](toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0) %}`

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+>    * Select the `RA__ThomasAM__species.csv` dataset imported into the history during the **Get the data** step;
+>    * Set a dimensionality of 10,000;
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:98]: You can use more icons here like `{% icon param-select %}`

Get some inspiration here for example: https://raw.githubusercontent.com/galaxyproject/training-material/main/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.md

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+>    * Set 100 levels;
+>    * Set 10 retraining iterations;
+>    * Set the number of folds to 5 to cross-validate the model;
+>    * Finally, disable the feature selection.
+>
+> The number of levels is the number of random vectors that `chopin2` generates for encoding encode data. This must be carefully selected since the accuracy of the resulting model is strictly correlated to this parameter ({% cite Cumbo2020 %}).
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:104]: encoding encode data?

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+> The tool will end up creating a `summary` file in your history containing a few basic information about the generated classification model and, more importantly, it's accuracy.
+>
+> We suggest you to repeat the same steps for building a classification model on the binary presence/absence profiles in `BIN__ThomasAM__species.csv`. In this case there is no need to use so many levels since the dataset contains only two possible values (i.e., 0 and 1). Thus, the number of levels must be changed to 2.
+>
+>    > <question-title></question-title>
+>    >
+>    > Compare the `summary` files generate by `chopin2` as the result of building a classification model over the `RA__ThomasAM__species.csv` and `BIN__ThomasAM__species.csv` datasets.
+>    > 1. Is there a difference in the accuracy of the two models?
+>    > 2. Does the difference in the number of levels have an impact on the running time and the final accuracy of the models?
+>    {: .question}
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:115]: Where are the solutions `> > <solution-title></solution-title>`

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+{: .hands_on}
+
+# Feature selection
+
+`chopin2` also implements a feature selection method based on the backward variable elimination strategy. It means that the tool will produce a classification model starting with the whole set of features in the dataset and iteratively remove those features that do not contribute to the accuracy of the model.
+
+Be aware that this specific type of feature selection method could lead to the generation of thousands of classification models in order to determine the best features and discard those ones that do not significantly contribute to a good accuracy. However, despite the huge amount of computational resources required to run the algorithm, it can easily handle datasets with massive amounts of features.
+
+> <hands-on-title>Identify the best features</hands-on-title>
+> **chopin2** {% icon tool %}: Run **chopin2** with the following configuration:
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:125]: tool annotation

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+>    * Select the `BIN__ThomasAM__species.csv` dataset imported into the history during the **Get the data** step;
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:126]: better param annotation with more icons

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+>
+> We are going to focus on the `BIN__ThomasAM__species.csv` dataset only this time since `chopin2` produced a classification model with a better accuracy compared to that of the model built over the `RA__ThomasAM__species.csv` dataset.
+>
+> The tool will end up creating a `summary` file in your history containing a line for each step of the backward variable elimination method. Every line corresponds to a classification model. Please note that the last column reports the features excluded in each step.
+>
+> A `selection` file will also appear in your history with the list of best features selected by `chopin2`.
+>
+>    > <question-title></question-title>
+>    >
+>    > Look at the list of microbial species (features) selected by `chopin2` reported in the `selection` file. Can you recognise any microbial species known to be linked in some way to the genesis and development of colorectal cancer?
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:141]: Answers are missing.

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+>    * Set 2 levels;
+>    * Set 10 retraining iterations;
+>    * Set the number of folds to 5 to cross-validate the model;
+>    * Finally, enable the feature selection.
+>
+> We are going to focus on the `BIN__ThomasAM__species.csv` dataset only this time since `chopin2` produced a classification model with a better accuracy compared to that of the model built over the `RA__ThomasAM__species.csv` dataset.
+>
+> The tool will end up creating a `summary` file in your history containing a line for each step of the backward variable elimination method. Every line corresponds to a classification model. Please note that the last column reports the features excluded in each step.
+>
+> A `selection` file will also appear in your history with the list of best features selected by `chopin2`.
```
Comment (bgruening) [inline topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:137]: A biological interpretation would be nice. What do we  see in the end comparing the stool datasets ...

File: topics/statistics/tutorials/hyperdimensional_computing/tutorial.md (hunk 3 above)
Comment (hexylena) [reply topics/statistics/tutorials/hyperdimensional_computing/tutorial.md:98]: I can recommend using PTDK on your workflow, and then copy pasting the "run tool" sections into this already quite far along tuotrial.

Comment (cumbof) [reply topics/statistics/tutorials/hyperdimensional_computing/tutorial.md]: I never used PTDK before. Thanks for the suggestion!

**Review by blankenberg — APPROVED** [review]

**Pull request merged**